src/apps/api/serializers/queues.py

```diff
@@ -15,28 +16,39 @@ class Meta:
         )
 
 
-class QueueCreationSerializer(serializers.ModelSerializer):
+class QueueOwnerMixin:
+    def get_is_owner(self, instance):
+        request = self.context.get('request')
+        if not request:
+            return None
+        return instance.owner and instance.owner == request.user
+
+
+class QueueCreationSerializer(QueueOwnerMixin, DefaultUserCreateMixin, serializers.ModelSerializer):
     is_owner = serializers.SerializerMethodField()
     organizers = serializers.PrimaryKeyRelatedField(queryset=User.objects.all(), many=True, required=False)
 
     class Meta:
         model = Queue
+        user_field = 'owner'
         fields = (
             'name',
-            'vhost',
             'is_public',
-            'owner',
             'organizers',
+            'owner',
             'broker_url',
+            'vhost',
             'created_when',
             'is_owner',
             'id',
         )
         read_only_fields = (
+            'owner',
             'broker_url',
             'vhost',
             'created_when',
             'is_owner',
+            'id',
```

Comment:
isn't 'id' read only by default? does it need to be included here?

Reply:
I'm not sure? have you poked at this and know it's read only?

Reply:
>The default implicitly-generated id field is marked as read_only.

from drf docs. could still manually test it though i suppose.

Reply:
cool, ty, will change ✔️